disk/disk_darwin.go

```diff
@@ -88,8 +91,60 @@ func getFsType(stat unix.Statfs_t) string {
 	return common.ByteToString(stat.Fstypename[:])
 }
 
+type SPNVMeDataTypeItem struct {
+	Name              string `json:"_name"`
+	BsdName           string `json:"bsd_name"`
+	DetachableDrive   string `json:"detachable_drive"`
+	DeviceModel       string `json:"device_model"`
+	DeviceRevision    string `json:"device_revision"`
+	DeviceSerial      string `json:"device_serial"`
+	PartitionMapType  string `json:"partition_map_type"`
+	RemovableMedia    string `json:"removable_media"`
+	Size              string `json:"size"`
+	SizeInBytes       int64  `json:"size_in_bytes"`
+	SmartStatus       string `json:"smart_status"`
+	SpnvmeTrimSupport string `json:"spnvme_trim_support"`
+	Volumes           []struct {
+		Name        string `json:"_name"`
+		BsdName     string `json:"bsd_name"`
+		Iocontent   string `json:"iocontent"`
+		Size        string `json:"size"`
+		SizeInBytes int    `json:"size_in_bytes"`
+	} `json:"volumes"`
+}
+
 func SerialNumberWithContext(ctx context.Context, name string) (string, error) {
-	return "", common.ErrNotImplementedError
+	cmd := exec.Command("system_profiler", "SPNVMeDataType", "-json")
+
+	output, err := cmd.Output()
+	if err != nil {
+		return "", err
+	}
+
+	var temp struct {
+		SPNVMeDataType []struct {
+			Items []SPNVMeDataTypeItem `json:"_items"`
+		} `json:"SPNVMeDataType"`
+	}
```

Comment:
How about defining the struct at the top level instead of dynamically defining it here?